src/main.py

```diff
@@ -308,6 +315,10 @@ def connect_to_device(self, host, port):
 
     def on_surface_changed(self):
         self.eye_tracking_provider.update_surface()
+        # TODO: is there a nicer way to implement this?
```

Comment:
Does this actually need to be re-set every time the surface changes?

Also, it seems like the `map_to_scene_video` function could be moved into the `TargetPresenter` class or `CalibrationWidget` (since that appears to be the only place it's used) if we made camera intrinsics part of the `EyeTrackingData` structure.

Reply:
No, it actually does not! The changes to the surface will result in different transformations, which will be given to `map_to_scene_video` as a parameter. The function itself will stay the same though. Good find!

Conceptually it doesn't fit so well into the calibration widgets and it should be closer to the surface tracking code, but I agree that the code simplification you'd get from putting this into the calibration widget would be nice and preferrable!

I'll change this!